From: Daniel Lehman <dlehman@esri.com>
Subject: [PATCH] msvcp140: Forward __ExceptionPtr* to msvcr120 (try2)
Message-Id: <c3420cceadc6482887c922a6b3935447@RED-INF-MXMB-P1.esri.com>
Date: Wed, 10 Aug 2016 16:50:57 +0000

Try2: simple forwards

From 1d56e226f319620553c1d903bdb6fcc5f6af46b2 Mon Sep 17 00:00:00 2001
From: Daniel Lehman <dlehman@esri.com>
Date: Thu, 16 Jun 2016 16:38:08 -0700
Subject: [PATCH] msvcp140: Forward __ExceptionPtr* to msvcr120

on older visual studios, the __ExceptionPtr* were in the C runtime
at vs2015, they were moved to the C++ runtime

Signed-off-by: Daniel Lehman <dlehman@esri.com>
---
 dlls/msvcp140/msvcp140.spec | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/dlls/msvcp140/msvcp140.spec b/dlls/msvcp140/msvcp140.spec
index 6e17e0d..f5a0838 100644
--- a/dlls/msvcp140/msvcp140.spec
+++ b/dlls/msvcp140/msvcp140.spec
@@ -1693,22 +1693,22 @@
 @ stub -arch=win64 ?__ExceptionPtrAssign@@YAXPEAXPEBX@Z
 @ stub -arch=win32 ?__ExceptionPtrCompare@@YA_NPBX0@Z
 @ stub -arch=win64 ?__ExceptionPtrCompare@@YA_NPEBX0@Z
-@ stub -arch=win32 ?__ExceptionPtrCopy@@YAXPAXPBX@Z
-@ stub -arch=win64 ?__ExceptionPtrCopy@@YAXPEAXPEBX@Z
+@ cdecl -arch=win32 ?__ExceptionPtrCopy@@YAXPAXPBX@Z(ptr ptr) msvcr120.?__ExceptionPtrCopy@@YAXPAXPBX@Z
+@ cdecl -arch=win64 ?__ExceptionPtrCopy@@YAXPEAXPEBX@Z(ptr ptr) msvcr120.?__ExceptionPtrCopy@@YAXPEAXPEBX@Z
 @ stub -arch=win32 ?__ExceptionPtrCopyException@@YAXPAXPBX1@Z
 @ stub -arch=win64 ?__ExceptionPtrCopyException@@YAXPEAXPEBX1@Z
-@ stub -arch=win32 ?__ExceptionPtrCreate@@YAXPAX@Z
-@ stub -arch=win64 ?__ExceptionPtrCreate@@YAXPEAX@Z
-@ stub -arch=win32 ?__ExceptionPtrCurrentException@@YAXPAX@Z
-@ stub -arch=win64 ?__ExceptionPtrCurrentException@@YAXPEAX@Z
-@ stub -arch=win32 ?__ExceptionPtrDestroy@@YAXPAX@Z
-@ stub -arch=win64 ?__ExceptionPtrDestroy@@YAXPEAX@Z
-@ stub -arch=win32 ?__ExceptionPtrRethrow@@YAXPBX@Z
-@ stub -arch=win64 ?__ExceptionPtrRethrow@@YAXPEBX@Z
+@ cdecl -arch=win32 ?__ExceptionPtrCreate@@YAXPAX@Z(ptr) msvcr120.?__ExceptionPtrCreate@@YAXPAX@Z
+@ cdecl -arch=win64 ?__ExceptionPtrCreate@@YAXPEAX@Z(ptr) msvcr120.?__ExceptionPtrCreate@@YAXPEAX@Z
+@ cdecl -arch=win32 ?__ExceptionPtrCurrentException@@YAXPAX@Z(ptr) msvcr120.?__ExceptionPtrCurrentException@@YAXPAX@Z
+@ cdecl -arch=win64 ?__ExceptionPtrCurrentException@@YAXPEAX@Z(ptr) msvcr120.?__ExceptionPtrCurrentException@@YAXPEAX@Z
+@ cdecl -arch=win32 ?__ExceptionPtrDestroy@@YAXPAX@Z(ptr) msvcr120.?__ExceptionPtrDestroy@@YAXPAX@Z
+@ cdecl -arch=win64 ?__ExceptionPtrDestroy@@YAXPEAX@Z(ptr) msvcr120.?__ExceptionPtrDestroy@@YAXPEAX@Z
+@ cdecl -arch=win32 ?__ExceptionPtrRethrow@@YAXPBX@Z(ptr) msvcr120.?__ExceptionPtrRethrow@@YAXPBX@Z
+@ cdecl -arch=win64 ?__ExceptionPtrRethrow@@YAXPEBX@Z(ptr) msvcr120.?__ExceptionPtrRethrow@@YAXPEBX@Z
 @ stub -arch=win32 ?__ExceptionPtrSwap@@YAXPAX0@Z
 @ stub -arch=win64 ?__ExceptionPtrSwap@@YAXPEAX0@Z
-@ stub -arch=win32 ?__ExceptionPtrToBool@@YA_NPBX@Z
-@ stub -arch=win64 ?__ExceptionPtrToBool@@YA_NPEBX@Z
+@ cdecl -arch=win32 ?__ExceptionPtrToBool@@YA_NPBX@Z(ptr) msvcr120.?__ExceptionPtrToBool@@YA_NPBX@Z
+@ cdecl -arch=win64 ?__ExceptionPtrToBool@@YA_NPEBX@Z(ptr) msvcr120.?__ExceptionPtrToBool@@YA_NPEBX@Z
 @ cdecl -arch=arm ?always_noconv@codecvt_base@std@@QBA_NXZ(ptr) codecvt_base_always_noconv
 @ thiscall -arch=i386 ?always_noconv@codecvt_base@std@@QBE_NXZ(ptr) codecvt_base_always_noconv
 @ cdecl -arch=win64 ?always_noconv@codecvt_base@std@@QEBA_NXZ(ptr) codecvt_base_always_noconv

-- 
1.9.5

